django/contrib/postgres/search.py

```diff
@@ -300,3 +303,90 @@ class TrigramSimilarity(TrigramBase):
 class TrigramDistance(TrigramBase):
     function = ''
     arg_joiner = ' <-> '
+
+
+class LexemeCombinable(Expression):
+    BITAND = '&'
+    BITOR = '|'
+
+    def _combine(self, other, connector, reversed, node=None):
+        if not isinstance(other, LexemeCombinable):
+            raise TypeError(
+                'Lexeme can only be combined with other Lexemes, '
+                'got {}.'.format(type(other))
+            )
+        if reversed:
+            return CombinedLexeme(other, connector, self)
+        return CombinedLexeme(self, connector, other)
+
+    # On Combinable, these are not implemented to reduce confusion with Q. In
+    # this case we are actually (ab)using them to do logical combination so
+    # it's consistent with other usage in Django.
+    def bitand(self, other):
+        return self._combine(other, self.BITAND, False)
+
+    def bitor(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __or__(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __and__(self, other):
+        return self._combine(other, self.BITAND, False)
+
+
+class Lexeme(LexemeCombinable, Value):
+    _output_field = SearchQueryField()
+
+    def __init__(self, value, output_field=None, *, invert=False, prefix=False, weight=None):
+        self.prefix = prefix
+        self.invert = invert
+        self.weight = weight
+        super().__init__(value, output_field=output_field)
+
+    def as_sql(self, compiler, connection):
+        param = "'%s'" % self.value.replace("'", "''").replace("\\", "\\\\")
```

Comment:
Using psycopg2's `adapt()` should be sufficient in all cases:
```suggestion
        param = psycopg2.extensions.adapt(self.value).getquoted().decode()
```

Reply:
Done, thanks!

Reply:
Getting a crash when using this with unicode characters:

```python
>>> psycopg2.extensions.adapt("L'amour piqué par une abeille").getquoted().decode()

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 14: invalid continuation byte
```